### PR TITLE
Migrate to SvelteKit release candidate

### LINKED ISSUE #502: Update Sveltekit Version
### What happened?

Update acmcsuf.com to latest sveltekit version in preparation for sveltekit v1.0

In relation to #500

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
Migration guide: <https://github.com/sveltejs/kit/discussions/5774>

### What this PR accomplishes

- Recreation of the website based on `npm create svelte@latest acmcsuf.com` (ran on 9/10/22)
- Moved [TypeScript] library/[Svelte] component files from `/src/lib/` to `/routes/` directory. Introduction of `/lib/public/legacy/` for extraneous files.
- Copied over docs, assets, and static content into `/` and `/static/`
- Renders the `/` index page
- Migrates `/src/service-worker/` (<https://kit.svelte.dev/docs/service-workers>)
- Migrates `/src/params/` (<https://kit.svelte.dev/docs/advanced-routing#matching>)
- Migrates error page, `/src/routes/+error.svelte` (<https://kit.svelte.dev/docs/routing#error>)
- Migrates Svelte hooks (<https://kit.svelte.dev/docs/hooks#server-hooks>)
- Disobeys the old file architecture

### What this PR consists of

This mega-PR consists of the following smaller PRs:

1. #540 Reviewed by @karnikaavelumani 
1. #541 Reviewed by @karnikaavelumani 
1. #554 Reviewed by @diamondburned 
1. #572 Reviewed by @karnikaavelumani 
1. #555 Reviewed by @erwinmedina 
1. #556 Reviewed by @Angus-1 
1. #558 Reviewed by @AlanCortez1337 
1. #559 Reviewed by @diamondburned 
1. #561 Reviewed by @Angus-1 
1. #568 Reviewed by @jaasonw 
1. #571 Reviewed by @jaasonw 
1. #573 Reviewed by @jaasonw 
1. #575 Reviewed by @Peekoe

Resolves #502.

### CODE REVIEW

**Comment by diamondburned** [comment]
LGTM.

File: src/lib/public/legacy/text-alignment.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum TextAlignment {
```
Comment (EthanThatOneKid) [inline src/lib/public/legacy/text-alignment.ts:1]: TextAlignment is a duplicate of `src/lib/public/text-alignment/text-alignment.ts`.

File: src/lib/server/blog/cache.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import type { BlogPost } from '$lib/public/blog/types';
+
+const EXPIRATION_TIMEOUT = 1e3 * 60 * 1; // Cache updates every 1 minute.
+
+interface BlogPostsCacheInterface {
+  setAllPosts(posts: BlogPost[]): void;
+  getAllPosts(): BlogPost[] | undefined;
+}
+
+export class BlogPostsCache implements BlogPostsCacheInterface {
```
Comment (EthanThatOneKid) [inline src/lib/server/blog/cache.ts:10]: Extend Cache<BlogPost[]> from `src/lib/server/cache/cache.ts`.

File: src/routes/events/[query].json/+server.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import type { RequestEvent } from '@sveltejs/kit';
+import type { RouteParams } from './$types';
+import { DEBUG_FLAG_ENABLED } from '$env/static/private';
```
Comment (EthanThatOneKid) [inline src/routes/events/[query].json/+server.ts:3]: Import DEBUG_FLAG_ENABLED from server flags.
```suggestion
import { DEBUG_FLAG_ENABLED } from '$lib/server/flags';
```

File: GETTING_STARTED.md [hunk 4]
```diff
@@ -2,26 +2,27 @@
 
 1. Install [Node 16 LTS](https://nodejs.org/en/)
 2. Install and get familiar with [Git](https://docs.github.com/en/get-started/quickstart/hello-world)
```
Comment (k4rni) [inline GETTING_STARTED.md:4]: We should probably provide a link to the actual download site for Git.

**Review by k4rni — APPROVED** [review]
Looks great!

File: src/routes/events/[query].json/+server.ts (hunk 3 above)
Comment (EthanThatOneKid) [reply src/routes/events/[query].json/+server.ts:3]: Followed up by #581.

File: src/lib/public/legacy/text-alignment.ts (hunk 1 above)
Comment (EthanThatOneKid) [reply src/lib/public/legacy/text-alignment.ts:1]: Followed up by #582.

File: src/lib/server/blog/cache.ts (hunk 2 above)
Comment (EthanThatOneKid) [reply src/lib/server/blog/cache.ts:10]: Followed up by #583.

File: GETTING_STARTED.md (hunk 4 above)
Comment (EthanThatOneKid) [reply GETTING_STARTED.md:4]: Followed up by #584.

File: src/routes/events/[query].json/+server.ts (hunk 3 above)
Comment (EthanThatOneKid) [reply src/routes/events/[query].json/+server.ts:3]: Resolved by #589.

**Pull request merged**